src/engine/TileManager.cxx

```diff
@@ -18,14 +19,8 @@ SDL_Texture *TileManager::getTexture(const std::string &id, size_t tileMapType)
 TileData *TileManager::getTileData(const std::string &id)
 {
   if (m_tileData.count(id))
-  {
     return &m_tileData[id];
-  }
-  else
-  {
-    LOG(LOG_ERROR) << "Non existing id is beeing accessed at Tile::getTileData(id) " << id;
-  }
-
+  throw UIError(TRACE_INFO "Non existing id is being accessed at " + string{__PRETTY_FUNCTION__} +  " => " + id);
```

Comment:
The unhandled exception that adct is encountering is happening on this line. The issue is that when a tile with the id of 'id' doesn't exist it used to produce an error log and then continue execution. This would happen a lot, since demolishion and a few other actions had no corresponding tile. Now it's throwing an exception, the exception is uncaught so its passing back up the call stack to main, then main terminates. 

We shouldn't be throwing an exception here unless we are going handle it. We will have this same problem in a whole heap of other places (basically everywhere that exceptions are being thrown where they weren't before) because we aren't really handling any exceptions at all. It's just manifesting itself here (right now) because it's somewhere where the normal operation is to throw the exception.

Reply:
If this happens often, then exceptions are not the way to go here. However, the exception should be caught and printed here from `main()`. 
- [ ] This exception should be caught and printed properly
- [ ] This should not throw an exception

Reply:
This should absolutely not be an exception. A debug level log is probably overkill, there isn't really much need for anything to be printed here, it's normal operation for the tile is to be empty. also, the main() catch-all error handler is a bit useless since it's in the main thread, it cannot catch exceptions thrown in the UI, game and soundloading threads (which are the ones that do basically everything). We probably need to put try-catches in the main game loop and the UI loop to catch most things.

Reply:
I think this code still runs in the main thread. As of now, nothing ever happens in the UI thread and the GameLoop thread only handles audio events. Also I'm pretty sure both of these threads already have a `try...catch` similar to main thread. Maybe exceptions are not casted properly...

Reply:
Fixed